feat: generated python-sdk

jarinox opened the pull request:
- To reduce development work on the python sdk it will be generated using swagger-codegen
- This PR includes the generated sdk merged together with the old sdk where old classes and methods have been marked deprecated. This makes the new sdk available while still providing a copy of the old sdk.

jarinox committed:
feat: add deprecation warnings to old python sdk

jarinox committed:
feat: add generated sdk

- sdk has been generated by swagger-codegen
- It is still possible to use the old sdk

jarinox committed:
feat: adapt test workflow for generated client

- ORS_API_KEY secret must be set in settings for workflow to work

koebi commented:
Hey,

 I played around with this for a bit, and out of the box, the following things are apparent:
 * until I have a client, I need 3 (from my perspective unintuitive) calls
 * I need to write my API key manually to a dict. typos here will lead to unclear errors.
 * I need a `DirectionsService` to request `DirectionsServiceApi`. This is both unclearly named and unintuitive
 * the return type of `get_geo_json_route` is an `InlineResponse2003` which is unintuitively named and no valid json, though the name suggests that.
 * The module exports around 200 different classes, which makes it very confusing and obfuscated
 
 The current module has 1 client class, 1 method/class per endpoint and a bit of util, so ~20 in total. That makes it rather clear to see what I want to use.
 
 With this, the client might get easier to maintain, but currently, it certainly doesn't get easier to use.
 To get any information, I still have to go through any sort of `route['features'][0]['properties']['summary']['duration']`, but the way to `route` is harder.

 Regarding the PR structure:
 * Documentation is in this PR, although another for documentation exists?

jarinox commented:
This PR contains the generated client and deprecation of the old client. The documentation contained in this PR is generated by swagger-codegen alongside the code.

The other PR contains vitepress configuration to host the generated docs.

jarinox commented:
The generated client in fact requires more code to achieve the same things. This was one of the first things I noticed and discussed when experimenting with the codegen.

Beside less maintenance the generated classes allow cleaner code when performing complex requests as the parameters are not specified as map but rather defined as objects.

Commit:
feat: regenerate using latest v7.1.1 docs

- Add snapping service
- Adapt MatrixServiceApi test to new client
- Remove unimplemented test templates

Commit:
feat: remove unused schemas

Commit:
feat: remove InlineResponse classes

Commit:
test: update tests to not use InlineResponse classes

Commit:
feat: add apiClient utility function

This function simlifies the creation of the generic ApiClient which is
used by the Service Apis.

Commit:
refactor: use apiClient method to configure client

The apiClient mehtod simplifies the creation of the ApiClient class.

Commit:
feat: make apiKey optional in apiClient helper

Commit:
refactor: examples to use most recent client

Commit:
Generated update. Please review before merging.

Commit:
Generated update. Please review before merging.

Commit:
refactor: add newline to trigger ci/cd

Commit:
feat: add newline to pyproject.toml

MichaelsJP committed:
feat: Integrate poetry into the pyproject.toml

MichaelsJP committed:
ci: Adjust the tox and the ci tests to use poetry

MichaelsJP committed:
fix: Fix wrong python version identifier

MichaelsJP committed:
chore: Update to latest ubuntu image

MichaelsJP committed:
chore: Remove python 3.7

MichaelsJP committed:
fix: Add pytet to allowlist

MichaelsJP committed:
chore: Remove old test requirements

MichaelsJP committed:
ci: Adjust the production pipeline

MichaelsJP committed:
chore: Remove unneeded files

MichaelsJP committed:
chore: Remove double checkout action

Commit:
feat: workflow for deploying to testpypi

Deploys on push to generated-client branch to testpypi

Commit:
build: rebuild client using latest templates

Commit:
test: deploy on testpypi on every commit

Commit:
Generated update. Please review before merging.

Commit:
Generated update. Please review before merging.

Commit:
Generated update. Please review before merging.

Commit:
Generated update. Please review before merging.

Commit:
Generated update. Please review before merging.

Commit:
Generated update. Please review before merging.

Commit:
Generated update. Please review before merging.

Commit:
Generated update for version 8.0.1. Please review before merging.

Commit:
Generated update for version 8.1.0. Please review before merging.

Commit:
Generated update for version 8.2.0. Please review before merging.

Commit:
Generated update for version 9.0.0. Please review before merging.
